src/main/groovy/org/transmartproject/batch/highdim/datastd/FilterDataWithoutAssayMappingsItemProcessor.groovy

```diff
@@ -0,0 +1,41 @@
+package org.transmartproject.batch.highdim.datastd
+
+import org.springframework.batch.core.configuration.annotation.StepScope
+import org.springframework.batch.item.ItemProcessor
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import org.transmartproject.batch.highdim.assays.AssayMappingsRowStore
+import org.transmartproject.batch.highdim.assays.MappingFileRow
+
+/**
+ * Filters out items (when the flag is on {@link this.skipUnmappedData}) that do not have assay mappings.
+ */
+@Component
+@StepScope
+class FilterDataWithoutAssayMappingsItemProcessor
```

Comment:
1) You're using both annotations here in the class and in the bean factory in the `@Configuration` class.
2) This class only needs to be job scoped.